src/app/tests/TestDefaultTermsAndConditionsProvider.cpp

```diff
@@ -0,0 +1,250 @@
+        .version = 1,
+    });
+
+    err = tncProvider.SetAcceptance(newTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+
+    chip::Optional<chip::app::TermsAndConditions> outTermsAndConditions;
+    err = tncProvider.GetAcceptance(outTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+    EXPECT_EQ(1, outTermsAndConditions.Value().value);
+    EXPECT_EQ(1, outTermsAndConditions.Value().version);
+
+    err = tncProvider.CommitAcceptance();
+    err = tncProvider.GetAcceptance(outTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+    EXPECT_EQ(1, outTermsAndConditions.Value().value);
+    EXPECT_EQ(1, outTermsAndConditions.Value().version);
+
+    err = anotherTncProvider.Init(&storageDelegate, termsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+
+    err = anotherTncProvider.GetAcceptance(outTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+    EXPECT_EQ(1, outTermsAndConditions.Value().value);
+    EXPECT_EQ(1, outTermsAndConditions.Value().version);
+}
+
+TEST(DefaultTermsAndConditionsProvider, TestTermsRequiredGetRequirementsSuccess)
+{
+    CHIP_ERROR err;
+
+    chip::TestPersistentStorageDelegate storageDelegate;
+    chip::app::DefaultTermsAndConditionsProvider tncProvider;
+
+    chip::Optional<chip::app::TermsAndConditions> termsAndConditions = chip::Optional<chip::app::TermsAndConditions>({
+        .value   = 1,
+        .version = 1,
+    });
+
+    err = tncProvider.Init(&storageDelegate, termsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+
+    chip::Optional<chip::app::TermsAndConditions> outTermsAndConditions;
+    err = tncProvider.GetRequirements(outTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+    EXPECT_EQ(1, outTermsAndConditions.Value().value);
+    EXPECT_EQ(1, outTermsAndConditions.Value().version);
+}
+
+TEST(DefaultTermsAndConditionsProvider, TestSetAcceptanceGetAcceptanceSuccess)
+{
+    CHIP_ERROR err;
+
+    chip::TestPersistentStorageDelegate storageDelegate;
+    chip::app::DefaultTermsAndConditionsProvider tncProvider;
+
+    chip::Optional<chip::app::TermsAndConditions> termsAndConditions = chip::Optional<chip::app::TermsAndConditions>({
+        .value   = 1,
+        .version = 1,
+    });
+
+    err = tncProvider.Init(&storageDelegate, termsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+
+    chip::Optional<chip::app::TermsAndConditions> acceptedTermsAndConditions = chip::Optional<chip::app::TermsAndConditions>({
+        .value   = 1,
+        .version = 1,
+    });
+    err = tncProvider.SetAcceptance(acceptedTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+
+    chip::Optional<chip::app::TermsAndConditions> outTermsAndConditions;
+    err = tncProvider.GetRequirements(outTermsAndConditions);
+    EXPECT_EQ(CHIP_NO_ERROR, err);
+    EXPECT_EQ(1, outTermsAndConditions.Value().value);
+    EXPECT_EQ(1, outTermsAndConditions.Value().version);
+}
+
+TEST(DefaultTermsAndConditionsProvider, TestRevertAcceptanceGetAcceptanceSuccess)
```

Comment:
Need a test for Set acceptance, commit, check then set, revert, get first commit.